src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -514,6 +529,31 @@ private int WriteRunToErrorList(Run run, string logFilePath)
             return sarifErrors.Count;
         }
 
+        private void ShowFilteredSuppressionStateColumn()
```

Comment:
>ShowFilteredSuppressionStateColumn [](start = 21, length = 34)

Can you please add a comment that explicitly describes the behavior we're looking for? i.e., in the event that we detect one or more suppressed items, we want to make the suppressions column visible and filter out suppressed items (so that they are not visible by default). This detailed comment will help readers validate whether the implementation meets expectations.

For example, if the suppressions column is *already* visible, and the user  has explicitly enabled viewing of suppressed items, we don't want to clear their setting. Can you confirm whether your code handles this scenario correctly? On surface it doesn't appear to.

Reply:
I added the comment. The code does handle this, in the `ColumnFilterer` class (which you probably hadn't read at the time you made the comment).

---
In reply to: [538746026](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538746026) [](ancestors = 538746026)